openmrs-19/src/main/java/org/motechproject/openmrs19/service/impl/OpenMRSRelationshipServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+package org.motechproject.openmrs19.service.impl;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.Collections2;
+import org.apache.commons.lang.StringUtils;
+import org.motechproject.openmrs19.config.Config;
+import org.motechproject.openmrs19.domain.Relationship;
+import org.motechproject.openmrs19.domain.RelationshipListResult;
+import org.motechproject.openmrs19.resource.RelationshipResource;
+import org.motechproject.openmrs19.service.OpenMRSConfigService;
+import org.motechproject.openmrs19.service.OpenMRSRelationshipService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service("relationshipService")
+public class OpenMRSRelationshipServiceImpl implements OpenMRSRelationshipService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OpenMRSRelationshipServiceImpl.class);
+
+    @Autowired
+    private OpenMRSConfigService configService;
+
+    @Autowired
+    private RelationshipResource relationshipResource;
+
+    @Override
+    public Relationship getByTypeUuidAndPersonUuid(String configName, String typeUuid, String personUuid) {
+        Config config = configService.getConfigByName(configName);
+
+        RelationshipListResult result = relationshipResource.getByPersonUuid(config, personUuid);
+
+        List<Relationship> relationships =  new ArrayList<>(Collections2.filter(result.getResults(), new Predicate<Relationship>() {
+            @Override
+            public boolean apply(Relationship relationship) {
+                String personBUuid = null;
+                String relationshipTypeUuid = null;
+
+                if (relationship.getPersonB() != null) {
+                    personBUuid = relationship.getPersonB().getUuid();
+                }
+
+                if (relationship.getRelationshipType() != null) {
+                    relationshipTypeUuid = relationship.getRelationshipType().getUuid();
+                }
+
+                return StringUtils.equals(personBUuid, personUuid)
```

Comment:
As mentioned in the [MOTECH-2356](https://applab.atlassian.net/browse/MOTECH-2356), at this moment some of the relationship filters doesn't work on the OpenMRS server. As a temporary workaround, we're filtering relationships by its "person", which includes both personA and personB fields. We're interested only in the personB, hence we are checking that person UUID.